io/shared/src/main/scala/fs2/io/net/Socket.scala

```diff
@@ -23,12 +23,12 @@ package fs2
 package io
 package net
 
-import com.comcast.ip4s.{IpAddress, SocketAddress}
+import com.comcast.ip4s.{GenSocketAddress, IpAddress, SocketAddress}
 import fs2.io.file.FileHandle
 
 /** Provides the ability to read/write from a TCP socket in the effect `F`.
   */
-trait Socket[F[_]] {
+trait Socket[F[_]] extends SocketInfo[F] {
```

Comment:
I think this is a breaking change, because `SocketInfo` brings abstract methods.